### ISSUE TITLE
Handle nullable (reference) types

### ISSUE THREAD

**Issue by robertsundstrom** [cezarypiatek/MappingGenerator#133]
If the property type is nullable (reference type), handle and generate code that checks for null.

Like with ```Image``` here:

```csharp
            return new Person()
            {
                Id = model.Id,
                Name = model.Name,
                Image = model.Image != null ? new ImageDto()
                {
                    Id = model.Image.Id,
                    Path = model.Image.Path,
                    Description = model.Image.Description,
                } : null
            };
```

**Comment by cezarypiatek** [comment]
How should I handle it when the destination is not nullable? Should I generate throw expression?

```cs
  return new Person()
            {
                Id = model.Id,
                Name = model.Name,
                Image = model.Image != null ? new ImageDto()
                {
                    Id = model.Image.Id,
                    Path = model.Image.Path,
                    Description = model.Image.Description,
                } : throw new NullArgumentException("model.Image")
            };
```

**Comment by robertsundstrom** [comment]
Yes. I think that it is an acceptable solution.

**Comment by cezarypiatek** [comment]
After a preliminary assessment, it looks like a quite complex feature. I started working on it but I can't promise when it will be delivered. In the meantime, please consider supporting MappingGenerator project ;)
![image](https://user-images.githubusercontent.com/7759991/86530539-cade1280-beb9-11ea-809e-d0f49c0a734c.png)

**Comment by robertsundstrom** [comment]
Well. I have bought you a coffee. 🙂  After having used AutoMapper lately, I feel direct code mappings is far more maintainable since it .

When I wrote this issue I was about to write a proposal on how to implement this but I opted not to.

You have to take the ``<Nullable>enable</Nullable>`` flag into consideration when generating the code.

Of the target is nullable (implicitly or explicitly with the flag) then there is no problem assigning from what could hold a null value.

What you want to generate is code that does not yield any warning or errors due to nullable.

**Comment by cezarypiatek** [comment]
I think that `<Nullable>enable</Nullable>` MSBuild flag doesn't matter here. The generated code should be shaped by nullability annotations (if they are available). The problem is a little bit more complex because I need to track all property accessors and apply additionally conditional accessor `.?` if it's needed. I'm in the middle of the implementation...

**Comment by cezarypiatek** [comment]
@robertsundstrom  Here's a pre-release version with the requested feature. Would you mind to test it and let me know if it works as expected? https://ci.appveyor.com/api/buildjobs/klonxsb0yjs8m3mi/artifacts/MappingGenerator%2FMappingGenerator%2FMappingGenerator.Vsix%2Fbin%2FRelease%2FMappingGenerator.vsix

**Comment by robertsundstrom** [comment]
@cezarypiatek OK. I will test it soon!

**Comment by cezarypiatek** [comment]
An official version with this feature has been released v1.17.435
